integration/pages/RosterTestPage.js

```diff
@@ -0,0 +1,155 @@
+    const hookState = Object.values(attendeeState)[0];
+    return hookState[property];
+  }
+
+  async getLengthOfAttendeeStates() {
+    const hookStates = this.find(ELEMENTS.hookStates);
+    const attendeeStates = await hookStates.findElements(ELEMENTS.attendeeState);
+    return attendeeStates.length;
+  }
+
+  async checkRosterLength(targetLength) {
+    const isMatched = await this.waitUntil(async () => {
+      const actualLength = await this.getRosterLength();
+      return actualLength === targetLength;
+    }, 5000);
+
+    assert(isMatched, `The length of roster is not ${targetLength}`);
+  }
+
+  async checkIfAttendeeIsInRoster(attendeeName) {
+    const isMatched = this.waitUntil(async () => {
+      const rosterCells = await this.findAll(ELEMENTS.rosterCell);
+      for (const rosterCell of rosterCells) {
+        const name = await rosterCell.getText();
+        return name === attendeeName;
+      }
+    }, 5000);
+
+    assert(isMatched, `Attendee ${attendeeName} is not in the roster`);
+  }
+
+  async checkLengthOfAttendeeStates(targetLength) {
+    const length = await this.getLengthOfAttendeeStates();
+    assert(length === targetLength, `The length of attendee states is not ${targetLength} but ${length}`);
+  }
+
+  async checkAttendeeState(attendeeName, property, value, equal = true) {
+    const passed = await this.waitUntil(async () => {
+      const state = await this.getAttendeeState(attendeeName, property);
+      if (equal) {
+        return state === value;
+      } else {
+        return state !== value;
+      }
+    }, 5000);
+
+    assert(passed, `Wrong attendee property, the ${property} of ${attendeeName} should be ${value}`);
+  }
+
+  async toggleMicrophone() {
+    await this.find(ELEMENTS.toggleMicrophoneButton).click();
+  }
+
+  async toggleVideo() {
+    await this.find(ELEMENTS.toggleVideoButton).click();
+  }
+
+  async toggleSpeaker() {
+    await this.find(ELEMENTS.toggleSpeakerButton).click();
+  }
+
+  async toggleContent() {
+    await this.find(ELEMENTS.toggleContentButton).click();
+  }
+
+  async checkMicrophoneIcon(attendeeName, muted) {
+    const isMatched = await this.waitUntil(async () => {
+      const rosterCell = await this.getAttendeeRosterCell(attendeeName);
+      const title = await rosterCell.findElement(By.css('title')).getAttribute('innerHTML');
+      const target = muted ? 'Muted microphone' : 'Microphone';
+      return title === target;
+    }, 5000);
+
+    assert(isMatched, `Wrong microphone icon when muted is ${muted}`);
+  }
+
+  async checkVideoIcon(attendeeName, videoEnabled) {
+    const isMatched = await this.waitUntil(async () => {
+      const rosterCell = await this.getAttendeeRosterCell(attendeeName);
+      const childElements = await rosterCell.findElements(By.xpath('*'));
+      const videoIcon = childElements[2];
+      const d = await videoIcon.findElement(By.css('path')).getAttribute('d');
+      return videoEnabled === d.startsWith('M19');
```

Comment:
What is M19 here? The video icons SVG I guess?
NIt: Request to add a comment.

Reply:
The only way that I can find to check the icon diff...
It's from the data of path define of the `<svg>`.

![image](https://user-images.githubusercontent.com/20471438/144116124-b0882ef1-ba2c-4efc-84d4-915f9c963447.png)

Reply:
Yup thought so.

Reply:
Added the comment.